cli-plugins/socket/socket_nodarwin.go

```diff
@@ -0,0 +1,19 @@
+//go:build !darwin
+
+package socket
+
+import (
+	"net"
+	"syscall"
+)
+
+func listen(socketname string) (*net.UnixListener, error) {
+	return net.ListenUnix("unix", &net.UnixAddr{
+		Name: "@" + socketname,
+		Net:  "unix",
+	})
+}
+
+func onAccept(conn *net.UnixConn, listener *net.UnixListener) {
+	syscall.Unlink(listener.Addr().String())
```

Comment:
Wasn't the socket getting automatically cleaned up on Linux but not on macOS? Since you are still using the namespaces socket on Linux, i.e., prefixing it with "@", won't it still get cleaned up? Shouldn't this unlink be in the socket_darwin onAccept?

Reply:
Shoot, good catch! Looks like I forgot that correction with the `git add -p`.

To test the behavior of this, it's sufficient to build a copy of the CLI and run a plugin command, and check that the sockets are no longer leaked.

To test the core functionality of the socket, something like `docker compose up` works (checking that after Ctrl-C, there is no detached docker-compose process still running).

Reply:
I saw @neersighted mention some reasoning behind this for Linux, but I agree that it's not obvious, so I would suggest adding a comment outlining this. (ISTR he had some references with details)

Reply:
LOL.. browsers and race-conditions... ignore my comment

Reply:
I'm not quite sure what you mean; the logic is Darwin-specific and not Linux/Windows-specific. Do you mean linking to the manpage for unix(7) where it is suggested that unlink(2) can be used to forward-remove a socket present in the filesystem?

Reply:
According to this https://docker.slack.com/archives/C0W4XJVFW/p1705067942173219?thread_ts=1705055721.030329&cid=C0W4XJVFW , Go will try to remove sockets not starting with "@" upon close. So perhaps the delete is not necessary in either case since no "@" is being prepended on darwin?

I was asking about automated tests…

Reply:
We're doing a forward removal so that even if the CLI process crashes/is killed, we don't leave a socket on "disk" (read: in $TMPDIR). You're right that it's not strictly needed if we exit gracefully, but we also don't need the socket to stick around longer than the first client connects as this is more-or-less a "handshake" or "rendezvous."

Regarding tests, we don't have any right now; I don't think it would be too terrible to integration test this, but we should leave that for a follow-up as this is a pretty painful regression and a blocker for 25.0.

Reply:
Also with regards to that manpage/the reason for doing the forward removal, the relevant text is here: https://man7.org/linux/man-pages/man7/unix.7.html#:~:text=Binding%20to%20a,it%20is%20closed.